### ISSUE TITLE
Allow user to select File System (Provider) when user selects `Save As`

### ISSUE THREAD

**Issue by DonJayamanne** [microsoft/vscode#109338]
* Assume an extension registers a file system provider `abc`.
* User is not using remote extension (as installing VSCode & its remote extension host on the remote machine is not possible).
* Assume user creates a file (using `File-> New File`).
* Clicking `Save As`, user is not presented with the option of saving the file on a remote file system.

VSCode assumes user is always dealing with local files.

It would be good for user or extension host to intercept this & present the user with the option of saving the file either locally or on the remote file system (if there are more, then user choses).
Basically:
* User will have to chose from the list of file system providers:
- local
- remote fs provider 1
- remote fs provider 2
etc..
